### DISCUSSION TITLE
ggml.c syntax errors when trying to build with cmake

### DISCUSSION THREAD

**Discussion by gamesaucer** [ggml-org/llama.cpp#952] [Q&A]
I'm following the instructions in the Readme exactly as specified. The `cmake ..` command worked just fine, but when I run `cmake --build . --config Release` I am immediately confronted with more than a hundred syntax errors, starting with `ggml.h(811,63): error C2146: syntax error: missing ')' before identifier 'x'` and ending with `ggml.c(10843,31): fatal error C1003: error count exceeds 100; stopping compilation`.

Any suggestions for what the issue might be?

**Comment by sw** [comment]
I assume it's the `GGML_RESTRICT`? Try removing all of them from the function `typedef`s. If that solves it, we should remove the `restrict` keyword entirely from the header file, I don't think it's necessary in the declaration.

**Reply by sw** [reply]
The error seems to occur when compiling ggml.c, so it's not due to C++ not having `restrict`. I suspect Cmake is not passing `/std:c11` to the compiler?

**Comment by wj267** [comment]
Removing the restrict keyword allowed compilation with cmake on windows for me.

**Reply by iHaagcom** [reply]
Can you elaborate please, im facing the same issue. adding #define restrict to ggml.cc allowed to get the some exe's but not the main.cpp

**Reply by wj267** [reply]
See the earlier response from sw. Open ggml.h in your editor and remove the `restrict` keyword from all the typedefs that use it. There are a lot of variables declared this way and I had to change them all.

**Reply by iHaagcom** [reply]
Thank you for the reply, looks like I cannot apply that same logic to this build for falcon: https://github.com/jploski/ggml/blob/master/include/ggml/ggml.h

**Reply by iHaagcom** [reply]
Looks like it’s something else for the falcon-ggml build
ggml.c(1442): fatal error c1003: error count exceeds 100

**Comment by torsionbear** [comment]
If you are using visual studio, check if it is newer than "Visual Studio 2019 version 16.8". /std::c11 is supported since then.
See https://learn.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-170.

CMake does not append the /std::c11 flag for older version of visual studio c compiler either.

**Comment by maksi65432** [comment]
the error is in the cmakelists and it the syntax error on new versions for me, it a bug
